### PR TITLE
Devnet Update Query

### PR DESCRIPTION
**Pull request by span14**
## Description
This PR creates a new `rpc.BlockNumber` type called `ConfirmedBlockNumber`, which serves to provide users access to the finalized block in V2.

### CODE REVIEW

File: eth/api_backend.go [hunk 1]
```diff
@@ -81,6 +82,22 @@ func (b *EthApiBackend) HeaderByNumber(ctx context.Context, blockNr rpc.BlockNum
...
+			current.Number,
+			current.Extra,
+			XDPoS.ExtraFieldCheck,
+		) == params.ConsensusEngineVersion2 {
+			// TO CHECK: why calling config in XDPoS is blocked (not field and method)
+			confirmedHash := b.XDPoS.EngineV2.GetLatestCommittedBlockInfo().Hash
+			return b.eth.blockchain.GetHeaderByHash(confirmedHash), nil
+		} else {
+			return nil, errors.New("PoS V1 does not support confirmed block loopup")
+		}
```
[the remark this answers is not in the document]
Comment (span14) [reply eth/api_backend.go:100]: Fixed

File: eth/api_backend.go [hunk 2]
```diff
@@ -81,6 +82,22 @@ func (b *EthApiBackend) HeaderByNumber(ctx context.Context, blockNr rpc.BlockNum
 	// Otherwise resolve and return the block
 	if blockNr == rpc.LatestBlockNumber {
 		return b.eth.blockchain.CurrentBlock().Header(), nil
+	} else if blockNr == rpc.ConfirmedBlockNumber {
+		if b.eth.chainConfig.XDPoS == nil {
+			return nil, errors.New("PoW does not support confirmed block loopup")
```
[the remark this answers is not in the document]
Comment (span14) [reply eth/api_backend.go]: Fix

**Review by wgr523 — APPROVED** [review]

**Review by wjrjerome — CHANGES_REQUESTED** [review]
it looks like the git commit is messed up. Your PR included a lot of my past commits.
Did you rebase or merge dev-upgrade into your branch?
Let's chat in zoom

**Review by wgr523 — CHANGES_REQUESTED** [review]
rebase needed

**Review by wjrjerome — APPROVED** [review]

**Pull request merged**